tests/integration/tests/test_util/snap.py

```diff
@@ -0,0 +1,87 @@
+#
+# Copyright 2024 Canonical, Ltd.
+#
+import json
+import logging
+import re
+import urllib.error
+import urllib.request
+from typing import List
+
+LOG = logging.getLogger(__name__)
+
+SNAP_NAME = "k8s"
+
+# For Snap Store API request
+SNAPSTORE_API = "https://api.snapcraft.io/v2/snaps/info/"
+SNAPSTORE_HEADERS = {
+    "Snap-Device-Series": "16",
+    "User-Agent": "Mozilla/5.0",
+}
+RISK_LEVELS = ["stable", "candidate", "beta", "edge"]
+
+
+def get_snap_info(snap_name=SNAP_NAME):
+    """Get the snap info from the Snap Store API."""
+    req = urllib.request.Request(SNAPSTORE_API + snap_name, headers=SNAPSTORE_HEADERS)
+    try:
+        with urllib.request.urlopen(req) as response:  # nosec
+            return json.loads(response.read().decode())
+    except urllib.error.HTTPError as e:
+        LOG.exception("HTTPError ({%s}): {%s} {%s}", req.full_url, e.code, e.reason)
+        raise
+    except urllib.error.URLError as e:
+        LOG.exception("URLError ({%s}): {%s}", req.full_url, e.reason)
+        raise
+
+
+def get_latest_channels(
+    num_of_channels: int, flavor: str, include_latest=True
+) -> List[str]:
+    """Get an ascending list of latest channels based on the number of channels and flavour.
+
+    e.g. get_latest_release_channels(3, "classic") -> ['1.31-classic/candidate', '1.30-classic/candidate']
+    if there are less than num_of_channels available, return all available channels.
+    Only the most stable risk level is returned for each major.minor version.
+    By default, the `latest/edge/<flavor>` channel is included in the list.
+    """
+    snap_info = get_snap_info()
+
+    # Extract channel information
+    channels = snap_info.get("channel-map", [])
+    available_channels = [ch["channel"]["name"] for ch in channels]
```

Comment:
arch awareness:

```suggestion
    available_channels = [ch["channel"]["name"] for ch in channels if ch["channel"]["architecture"] == arch]
```

Reply:
Thanks TIL that channels can be arch specific.